WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -719,22 +772,119 @@ private extension EditableOrderViewModel {
             .assign(to: &$statusBadgeViewModel)
     }
 
+    /// Creates an array of OrderSyncProductInput that will be sent to the RemoteOrderSynchronizer when adding multiple products to an Order
+    /// - Parameters:
+    ///   - products: Selected products
+    ///   - variations: Selected product variations
+    /// - Returns: [OrderSyncProductInput]
+    ///
+    func productInputAdditionsToSync (products: [Product?], variations: [ProductVariation?]) -> [OrderSyncProductInput] {
+        var productInputs: [OrderSyncProductInput] = []
+        var productVariationInputs: [OrderSyncProductInput] = []
+
+        // TODO: We may be able to simplify this bit by using createUpdateProductInput() helper
```

Comment:
I don't think we can use `createUpdateProductInput` here, because it won't add products which aren't already in the order, but definitely using the `order.items` as the source of truth will help.

Reply:
Yeah, I though we could use it initially but is meant to update an existing input from `OrderSynchronizer`, which we do not have yet when creating the product inputs. As you suggested, I've used `order.items` instead to get the latest order sync when creating those: https://github.com/woocommerce/woocommerce-ios/pull/9203/commits/ee49133f0600c0b8727acff8d8f0b62b07b49afd